### ISSUE TITLE
Unspecified data type in image_dataset_from_directory

### ISSUE THREAD

**Issue by AngCamp** [tensorflow/tensorflow#52336]
Thank you for submitting a TensorFlow documentation issue. Per our GitHub
policy, we only address code/doc bugs, performance issues, feature requests, and
build/installation issues on GitHub.

The TensorFlow docs are open source! To get involved, read the documentation
contributor guide: https://www.tensorflow.org/community/contribute/docs

## URL(s) with the issue: https://keras.io/api/preprocessing/image/#imagedatasetfromdirectory-function

There are issues with the follwoing entries: 

**labels:** Either "inferred" (labels are generated from the directory structure), None (no labels), or a list/tuple of integer labels of the same size as the number of image files found in the directory. Labels should be sorted according to the alphanumeric order of the image file paths (obtained via os.walk(directory) in Python).
**label_mode:** - 'int': means that the labels are encoded as integers (e.g. for sparse_categorical_crossentropy loss). - 'categorical' means that the labels are encoded as a categorical vector (e.g. for categorical_crossentropy loss). - 'binary' means that the labels (there can be only 2) are encoded as float32 scalars with values 0 or 1 (e.g. for binary_crossentropy). - None (no labels).

## Description of issue (what needs changing):

The data types for input are not clearly described and there are no examples of what would be an acceptable input if its is not feasible to supply a data directory organized into files by class labels.  Currently when i try this approach the fucntion does not even recognize that there are files in the directory I am giving it, but when I move up in the directory and supply the file containing the images instead I get a Dataset object for training a single class. 

The errors are not useful and confusing.  It is not clear why the function does not recognize that there are image files in the folder in one case and not another.

**Comment by mohantym** [comment]
Hi @AngCamp! 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues) 
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!
